client/state/sharing/keyring/schema.js

```diff
@@ -0,0 +1,29 @@
+export const itemSchema = {
+	type: 'object',
+	patternProperties: {
+		'^\\d+$': {
+			type: 'object',
+			required: 'ID',
+			properties: {
+				ID: { type: 'integer' },
+				additional_external_users: { type: 'array' },
+				expires: { type: 'string' },
+				external_ID: { type: [ 'string', 'null' ] },
+				external_display: { type: [ 'string', 'null' ] },
+				external_name: { type: [ 'string', 'null' ] },
+				external_profile_picture: { type: [ 'string', 'null' ] },
+				issued: { type: 'string' },
+				label: { type: 'string' },
+				meta: { type: 'object' },
+				refresh_URL: { type: 'string' },
+				service: { type: 'string' },
+				sites: { type: 'array' },
+				status: { type: 'string' },
+				type: { type: 'string' },
+				user_ID: { type: 'integer' },
+			},
+			additionalProperties: false
```

Comment:
I don't believe we need to be so strict here. We might have an additive change to the api that isn't yet used by the client. In that case this would throw out the persisted data needlessly.